### DISCUSSION TITLE
Events when adding a new column/field/filter/button to the crud

### DISCUSSION THREAD

**Discussion by susanu** [Laravel-Backpack/community-forum#699] [General] [answered]
Hi,

In my project, all enum classes have a "label" function and i have to define 'enum_function' to all enum type columns and fields in order to use that function.
I know i can use macros but i was wondering if it's useful to have some sort of events for column/fields/button/filters.

Thank you!

**Comment by jcastroa87** [comment]
Hello @susanu 

At the moment we do not have that functionality in Backpack and we do not have a plan for it in our roadmap, however, we are always attentive to the needs of our users and we will evaluate them depending on the demand and impact it has.

The closest function that can listener to events is our [Javascript Api](https://backpackforlaravel.com/docs/6.x/crud-fields-javascript-api#how-to-use-1) and as you say the more suitable one is handled with macros.

Cheers.

**Comment by jcastroa87** [comment]
Hello @susanu i hope you are doing well.

Please let us know if you have any other requests or we can close this discussion.

Thanks.

**Comment by susanu** [comment, marked as answer]
Hi @jorgetwgroup,

Although the events are a nice thing to have or some sort of default values for specific column/fields types i will go ahead and implement this using the macros functionality.

Thank you!

**Comment by jcastroa87** [comment]
Thanks to you @susanu
